setup/Setup.CLI/Commands/BaseCommand.cs

```diff
@@ -0,0 +1,76 @@
+using Spectre.Console;
+using Terraria.ModLoader.Setup.Core;
+
+namespace Terraria.ModLoader.Setup.CLI.Commands;
+
+public abstract class BaseCommand
+{
+	private readonly TargetsFilesUpdater targetsFilesUpdater;
+	private readonly TerrariaExecutableSetter terrariaExecutableSetter;
+
+	protected BaseCommand(
+		ProgramSettings programSettings,
+		TargetsFilesUpdater targetsFilesUpdater,
+		TerrariaExecutableSetter terrariaExecutableSetter)
+	{
+		this.ProgramSettings = programSettings;
+		this.targetsFilesUpdater = targetsFilesUpdater;
+		this.terrariaExecutableSetter = terrariaExecutableSetter;
+	}
+
+	protected ProgramSettings ProgramSettings { get; }
+
+	protected async Task RunTask(SetupOperation task, bool plainProgress, bool noPrompts = false, CancellationToken cancellationToken = default)
+	{
+		var errorLogFile = Path.Combine(ProgramSettings.LogsDir, "error.log");
+
+		try {
+			SetupOperation.DeleteFile(errorLogFile);
+
+			await terrariaExecutableSetter.FindAndSetTerrariaDirectoryIfNecessary(noPrompts, cancellationToken);
+			targetsFilesUpdater.Update();
+
+			if (!await task.ConfigurationPrompt(cancellationToken))
+				return;
+
+			if (!task.StartupWarning())
+				return;
+
+			if (plainProgress) {
+				await task.Run(new PlainConsoleProgress(), cancellationToken);
+			}
+			else {
+				var table = new Table()
+					.HideRowSeparators()
+					.ShowHeaders()
+					.NoBorder()
+					.Expand()
+					.AddColumn(string.Empty)
+					.AddEmptyRow();
+
+				await AnsiConsole
+					.Live(table)
+					.StartAsync(async ctx => await task.Run(new LiveConsoleProgress(ctx, table), cancellationToken));
+			}
+
+			if (task.Failed() || task.Warnings()) {
+				task.FinishedPrompt();
+			}
+
+			Text text = task.Failed()
+				? new Text("Failed", new Style(foreground: Color.Red, decoration: Decoration.Bold))
+				: new Text("Done", new Style(foreground: Color.Green, decoration: Decoration.Bold));
+
+			AnsiConsole.Write(text);
+		}
+		catch (OperationCanceledException) { }
+		catch (Exception exception) when (exception is not OperationCanceledException) {
+			AnsiConsole.WriteException(exception, ExceptionFormats.NoStackTrace);
+
+			SetupOperation.CreateDirectory(ProgramSettings.LogsDir);
+			await File.WriteAllTextAsync(errorLogFile, exception.ToString(), cancellationToken);
```

Comment:
Need to print a message which directs people to log file path. (We should do this in the GUI message box too)

Reply:
Added to both